source/geometries/Next100FieldCage.cc

```diff
@@ -613,6 +656,149 @@ void Next100FieldCage::BuildFieldCage()
   }
 }
 
+void Next100FieldCage::BuildFieldCage()//////////////////////////////////////////
+{
+  // HDPE cilinder.
+  G4double hdpe_tube_z_pos = -hdpe_length_/2. + teflon_total_length_ + gate_teflon_dist_ + GetELzCoord();
+
+  G4Tubs* hdpe_tube_solid =
+    new G4Tubs("HDPE_TUBE", hdpe_tube_int_diam_/2.,
+               hdpe_tube_ext_diam_/2., hdpe_length_/2., 0, twopi);
+
+  G4LogicalVolume* hdpe_tube_logic = new G4LogicalVolume(hdpe_tube_solid,
+                                                         hdpe_, "HDPE_TUBE");
+
+  new G4PVPlacement(0, G4ThreeVector(0., 0., hdpe_tube_z_pos), hdpe_tube_logic,
+                                     "HDPE_TUBE", mother_logic_, false, 0, true);
+
+  G4double active_short_z = 13.5*mm; //Thickness for the small piece from the  holder in front of the ring.
+  G4double first_ring_drif_z_pos = drift_ring_dist_/2. + active_short_z/2. + gate_teflon_dist_ + GetELzCoord();
```

Comment:
I defined `first_ring_drif_z_pos` like that to avoid possible infinitesimal overlaps because I wasn't sure if the space holding the rings (marked in yellow in the image) was exactly equal to the `ring_thickn_`
<img width="743" alt="first_ring_drif_z_pos" src="https://user-images.githubusercontent.com/44235769/127535267-243862bd-66bd-490a-a0c4-fb5c6aca8a3f.png">

Reply:
Ok, I see, it is clear now. I hadn't realised that the `ring_thickn_` is not equal to the spacing where they are positioned.